src/m365/planner/commands/task/task-checklistitem-remove.ts

```diff
@@ -0,0 +1,111 @@
+import { Cli, Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id: string;
+  taskId: string;
+  confirm?: boolean;
+}
+
+class PlannerTaskChecklistItemRemoveCommand extends GraphCommand {
+  public get name(): string {
+    return commands.TASK_CHECKLISTITEM_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes the checklist item from the planner task';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.confirm = (!(!args.options.confirm)).toString();
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (args.options.confirm) {
+      this.removeChecklistitem(logger, args, cb);
+    }
+    else {
+      Cli.prompt({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove the checklist item with id ${args.options.id} from the planner task?`
+      }, (result: { continue: boolean }): void => {
+        if (!result.continue) {
+          cb();
+        }
+        else {
+          this.removeChecklistitem(logger, args, cb);
+        }
+      });
+    }
+  }
+
+  private removeChecklistitem(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    this
+      .getTaskDetailsEtag(args.options)
+      .then(etag => {
+        const requestOptionsTaskDetails: any = {
+          url: `${this.resource}/v1.0/planner/tasks/${args.options.taskId}/details`,
+          headers: {
+            'accept': 'application/json;odata.metadata=none',
+            'If-Match': etag,
+            'Prefer': 'return=representation'
+          },
+          responseType: 'json',
+          data: {
+            checklist: {
+              [args.options.id]: null
+            }
+          }
+        };
+
+        return request.patch(requestOptionsTaskDetails);
+      })
+      .then(_ => cb(), (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getTaskDetailsEtag(options: Options): Promise<string> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/tasks/${encodeURIComponent(options.taskId)}/details`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get(requestOptions)
+      .then((response: any) => {
+        const etag: string = response['@odata.etag'];
+        const checklistItemId = response.checklist[options.id];
+        if (!checklistItemId) {
```

Comment:
You are now doing two 'unrelated' things in a function that (according to it's name) should only do one thing: return a task etag. For clarities sake, I'd like to do the checklistitem validation outside of this function. You could rename `getTaskDetailsEtag` to `getTaskDetails`, and just return the entire task. Then validate the checklistItem before you create the `requestOptionsTaskDetails` request object.

Reply:
@martinlingstuyl - Updated the code, can you please review? Much appreciated for your review